src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -841,6 +983,11 @@ private static XElement GetSection(XElement parentElement, string section)
             return parentElement.Element(section);
         }
 
+        private static IEnumerable<XElement> GetAllSections(XElement parentElement)
+        {
+            return parentElement.Elements();
```

Comment:
I'm slightly confused by this.  [Earlier](https://github.com/NuGet/NuGet.Client/pull/2021/files#diff-86990f2adab67354828de3f4f08b388cR523) `GetAllSubsections(...)` walks the `ISettings` hierarchy to collect and return all subsections.  This method (`GetAllSections(...)`) appears to not walk the `ISettings` hierarchy.  Am I missing something or are these two methods inconsistent in use of `GetAll...`?